src/app/providers.tsx

```diff
@@ -0,0 +1,39 @@
+'use client';
+import { ReactNode } from 'react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
+import { ReactQueryStreamedHydration } from '@tanstack/react-query-next-experimental';
+
+function makeQueryClient() {
+  return new QueryClient({
+    defaultOptions: {
+      queries: {
+        staleTime: 60 * 1000,
+      },
+    },
+  });
+}
+
+let browserQueryClient: QueryClient | undefined = undefined;
+
+function getQueryClient() {
+  if (typeof window === 'undefined') {
+    return makeQueryClient();
+  } else {
+    if (!browserQueryClient) browserQueryClient = makeQueryClient();
+    return browserQueryClient;
+  }
+}
+
+export function Providers(props: { children: ReactNode }) {
+  const queryClient = getQueryClient();
+
+  return (
+    <QueryClientProvider client={queryClient}>
+      <ReactQueryStreamedHydration>
+        {props.children}
+      </ReactQueryStreamedHydration>
+      <ReactQueryDevtools initialIsOpen={false} />
+    </QueryClientProvider>
+  );
+}
```

Comment:
P4: 구조분해 할당 해도 좋을거 같습니당 
p4: 컴포넌트가 props로 `children` 1개만 받을땐 PropsWithChildren유틸리티 타입 쓰는 방법도 있습니다!
근데 이제 찾아보니까 옵셔널할때만 쓰는게 좋다고 하니 요건 참고만 해주세용
```suggestion
export function Providers({ children }: PropsWithChildren}) {
  const queryClient = getQueryClient();

  return (
    <QueryClientProvider client={queryClient}>
      <ReactQueryStreamedHydration>
       {children}
      </ReactQueryStreamedHydration>
      <ReactQueryDevtools initialIsOpen={false} />
    </QueryClientProvider>
  );
}
```
++ 프롭스 한개여도 인터페이스로 뺄건지 회의때 이야기 하면 좋을거 같아요!
```
interface ProvidersProps {
  children: ReactNode;
}

export function Providers({ children }: ProvidersProps) {
 
}
```

Reply:
해당 내용은 공식문서에서 발췌해온 내용이라 저희가 사용하던 일방적인 방식과는 살짝 갭이 있던 부분이네요
작성해주신대로 수정하겠습니다~

Reply:
props가 한개라도 위 방식디로 interface로 선언하는게 일관성있고 좋다고 생각합니다